TrackletAlgorithm/VMRouter.h

```diff
@@ -229,26 +214,31 @@ inline VMStubME<OutType> createStubME(const InputStub<InType> stub,
 		// Stubs can only end up in the neighbouring VM after calculating iphivmrawplus/minus
 	assert(std::abs(ivmMinus - ivmPlus) <= 1);
 
+
 	// Indices used to find the rzfine value in finebintable
 	// finebintable returns the top 6 bits of a corrected z
 	// Note: not the index that is being saved to the stub
 	ap_uint<nbitszfinebintable + nbitsrfinebintable> indexz = ((z
 			+ (1 << (nzbits - 1))) >> (nzbits - nbitszfinebintable)); // Make z unsigned and take the top "nbitszfinebintable" bits
-	ap_uint<nbitsrfinebintable> indexr = 0;
+	ap_uint<nbitsrfinebintable> indexr = -1;
+
+	constexpr int rbins = (1 << nbitsrfinebintable); // Number of bins in r in finebintable
 
 	if (Disk) {
 		if (negdisk) {
 			indexz = (1 << nbitszfinebintable) - indexz;
 		}
-		indexr = r >> (nrbits - nbitsrfinebintable); // Take the top "nbitsrfinebintable" bits
+		indexr = r;
+		if (InType == DISKPS) {
+			indexr = r >> (nrbits - nbitsrfinebintable); // Take the top "nbitsrfinebintable" bits
+		}
 	} else { // Layer
 		indexr = ((r + (1 << (nrbits - 1)))
 				>> (nrbits - nbitsrfinebintable)); // Make r unsigned and take the top "nbitsrfinebintable" bits
 	}
 
 	// The index for finebintable
-	ap_uint<nbitszfinebintable + nbitsrfinebintable> finebinindex = (indexz
-			<< nbitsrfinebintable) + indexr;
+	ap_uint<nbitszfinebintable + nbitsrfinebintable> finebinindex = (indexz * rbins) + indexr;
 
 	// Get the corrected r/z position
 	ap_uint<6> rzcorr = finebintable[finebinindex];
```

Comment:
change to `auto` to get rid of magic number?

Reply:
Done.